src/commands/generate/utils.ts

```diff
@@ -91,3 +90,18 @@ const strictModeReservedKeywords = new Set([
     "static",
     "yield",
 ]);
+
+type DeprecatableDefinitions = IFieldDefinition | IEnumValueDefinition | IEndpointDefinition;
+export function addDeprecatedToDocs<T extends DeprecatableDefinitions>(typeDefintion: T): T {
```

Comment:
can we avoid mutating objects and instead construct a better doc string when creating the object?

Reply:
Changed so it doesn't mutate. It is a little weird how before though the docs were being put on as a JSDocStructure when a string will do. I also saw it took an array so tried to pass it in with 2 strings, but the second would not be picked up. I'm guessing it's just old code so just changed to pass back a string[] or undefined.